### PR TITLE
the API documentation for f5lbaasdriver.utils is broken

### PR DESCRIPTION
**Pull request by mattgreene**
LBaaSv2 solutions test has broken graphics links

Fixes:
Issue #524
Issue #480

Problem:
Mulitple doc problems including: missing sections and broken links.

Analysis:
- The requirements.doc.txt file was not pinned to the correct openstack distro to match the version of docs being generated.  Pinning it correctly introduces the need to manually download the F5 driver shim and install into the virtual env used to generate docs.
- The driver is used solely by the neutron_lbaas.  There is no need to generate "API" documentation, since it is not a valid use-case to directly interact with the driver python code.
- Fixed a few easy doc warnings along with image links pointing to the wrong directory.
- Created a local upper-constraints file ONLY for docs that clones the upstream community but removes the constraint for sphinx.  We need to use whatever version of sphinx we want to generate product docs, but need the constraints file to make sure release-appropriate packages are installed in the virtual env since generating the docs relies on importing neutron and neutron_lbaas.

Tests:
sphinx-build in local environment with inspection of generated docs.

### CODE REVIEW

**Comment by mattgreene** [comment]
These changes won't work against the default RTD build environment.  RTD supports a DOCKER_IMAGE flag to pick the build env container, but not sure how to override the setting in hosted RTD.

**Comment by mattgreene** [comment]
Updated this PR to exclude version-pinning changes.  This part cannot be solved until we migrate to clouddocs.

File: docs/solution-test-plan.rst [hunk 1]
```diff
@@ -81,7 +81,7 @@ NA1: Global Routed Mode
 
 Edge deployment architecture using only OpenStack networking provider networks, with F5 OpenStack agents deployed in :ref:`Global Routed Mode <global-routed-mode>`.
 
-.. figure:: ../media/f5-lbaas-test-architecture-grm.png
+.. figure:: media/f5-lbaas-test-architecture-grm.png
```
Comment (jputrino) [inline docs/solution-test-plan.rst]: I think this might have to be `/media/`

File: docs/solution-test-plan.rst [hunk 2]
```diff
@@ -90,7 +90,7 @@ NA2: L2 Adjacent Mode
 
 Micro-segmentation architecture using tenant networks, with F5 agents deployed in :ref:`L2 Adjacent Mode <L2 Segmentation Mode>`. Execute tests for VLAN and then VxLAN network types.
 
-.. figure:: ../media/f5-lbaas-test-architecture-l2adj.png
+.. figure:: media/f5-lbaas-test-architecture-l2adj.png
```
Comment (jputrino) [inline docs/solution-test-plan.rst]: same as above -- `/media/`

Comment (jputrino) [inline docs/solution-test-plan.rst]: I tested the docs build and the images are there, so this is good to go.

**Pull request merged**